mqtt/src/main/scala/Mqtt.scala

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.contrib.mqtt
+
+import akka.util.ByteString
+import akka.stream.stage._
+import org.eclipse.paho.client.mqttv3.{ MqttMessage => PahoMqttMessage, _ }
+import scala.util._
+import scala.language.implicitConversions
+
+case class MqttSourceSettings(
+  connectionSettings: MqttConnectionSettings,
+  topics:             Map[String, Int]
+)
+
+case class MqttConnectionSettings(
+  broker:      String,
+  clientId:    String,
+  persistence: MqttClientPersistence
+)
+
+case class MqttMessage(topic: String, payload: ByteString)
+
+trait MqttConnectorLogic { this: GraphStageLogic =>
+
+  import MqttConnectorLogic._
+
+  def connectionSettings: MqttConnectionSettings
+  def onConnect: AsyncCallback[IMqttAsyncClient]
+  def onMessage: AsyncCallback[MqttMessage]
+  def onConnectionLost: AsyncCallback[Throwable]
+
+  final override def preStart(): Unit = {
+    val client = new MqttAsyncClient(
+      connectionSettings.broker,
+      connectionSettings.clientId,
+      connectionSettings.persistence
+    )
+
+    client.connect((), connectHandler(client))
+    client.setCallback(new MqttCallback {
+      def messageArrived(topic: String, message: PahoMqttMessage) =
+        onMessage.invoke(MqttMessage(topic, ByteString(message.getPayload)))
+
+      def deliveryComplete(token: IMqttDeliveryToken) =
+        println(s"Delivery complete $token")
+
+      def connectionLost(cause: Throwable) =
+        onConnectionLost.invoke(cause)
+    })
```

Comment:
Isn't there a race here, connect happens before `setCallback` or does the client not accept messages before callback is set?